src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -120,10 +147,132 @@ class DelimiterStatement(BaseSegment):
     match_grammar = Ref.keyword("DELIMITER")
 
 
+@mysql_dialect.segment()
+class CreateProcedureStatementSegment(BaseSegment):
+    """A `CREATE PROCEDURE` statement.
+
+    mysql: https://dev.mysql.com/doc/refman/8.0/en/create-procedure.html
+    """
+
+    type = "create_procedure_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        "PROCEDURE",
+        Anything(),
+    )
+
+    parse_grammar = Sequence(
+        "CREATE",
+        "PROCEDURE",
```

Comment:
I actually left it out on purpose.  I spent significant time in trying to get it to parse the DEFINER statement but it wasn't working correctly.  So I left it out so I could open a new issue to address it later since that statement is optional.

Reply:
got it! could we open that issue already and refer to it here?

Reply:
I created an issue for this, but I have already started a bunch of new stuff to expand on the mysql dialect.  Instead of reverting all my changes to resolve the issues with the `CHANGELOG` and this documentation item.  Can we proceed with this how it is and get it merged and I will clean up these items in my next PR?

Reply:
@silverbullettruck2001 sorry but I'm not in favor of leaving the CHANGELOG in an unclean state for even a few commits because it might cause further issues/confusions on other PRs

blocking the changes now explicitly (since the comments above were marked as resolved)